### PR TITLE
Notification base

### PR DESCRIPTION
**Pull request by VaclavK**

### CODE REVIEW

**Comment by VaclavK** [comment]
@djfr you have seen it - PR created so that we following the process - even if we do offline reviews on occasions - this is "recorded"

File: src/Eshopworld.Core/BaseNotification.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿namespace Eshopworld.Core
+{
+    /// <summary>
+    /// base type for all notifications
+    /// </summary>
+    public class BaseNotification
```
Comment (arturek) [inline src/Eshopworld.Core/BaseNotification.cs]: abstract?

File: src/Eshopworld.Core/BaseNotification.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    {
+        /// <summary>
+        /// non public constructor to prevent instantiation at this level
+        /// </summary>
+        protected BaseNotification()
```
Comment (arturek) [inline src/Eshopworld.Core/BaseNotification.cs]: internal?

**Review by arturek — APPROVED** [review]

**Comment by djfr** [comment]
Made the changes suggested by @arturek because it allows for that base class to be mocked and achieves the same level of external encapsulation as before. Also bumped minor instead of revision.

**Review by djfr — APPROVED** [review]

**Pull request merged**